src/main/java/sx/blah/discord/util/MissingPermissionsException.java

```diff
@@ -10,23 +10,28 @@
  */
 public class MissingPermissionsException extends Exception {
 
-	private EnumSet<Permissions> missing;
+	private final EnumSet<Permissions> missing;
 
 	public MissingPermissionsException(EnumSet<Permissions> permissionsMissing) {
 		super(getMessage(permissionsMissing));
 		missing = permissionsMissing;
 	}
 
-	public MissingPermissionsException(String message) {
-		super(message);
-	}
-
 	private static String getMessage(EnumSet<Permissions> permissions) {
 		StringJoiner joiner = new StringJoiner(", ");
 		permissions.stream()
-				.map(p -> p.name())
-				.forEach(p -> joiner.add(p));
-		return "Missing permissions: "+joiner.toString()+"!";
+				.map(Enum::name)
+				.forEach(joiner::add);
+		return "Missing permissions: " + joiner.toString() + "!";
+	}
+
+	/**
+	 * Gets the missing permissions.
+	 *
+	 * @return The permissions.
+	 */
+	public EnumSet<Permissions> getMissing() {
+		return missing;
```

Comment:
Just make it public and wrap it in Collections.immutableSet or the guava ImmutableEnumSet